mava/systems/q_learning/anakin/rec_qmix.py

```diff
@@ -0,0 +1,697 @@
+    init_x = (init_obs_batched, init_term_or_trunc)
+    # (B, A, ...)
+    init_hidden_state = ScannedRNN.initialize_carry(
+        (cfg.arch.num_envs, num_agents), cfg.network.hidden_state_dim
+    )
+
+    # Making recurrent Q network
+    pre_torso = hydra.utils.instantiate(cfg.network.q_network.pre_torso)
+    post_torso = hydra.utils.instantiate(cfg.network.q_network.post_torso)
+    q_net = RecQNetwork(
+        pre_torso=pre_torso,
+        post_torso=post_torso,
+        num_actions=action_dim,
+        hidden_state_dim=cfg.network.hidden_state_dim,
+    )
+    q_params = q_net.init(q_key, init_hidden_state, init_x)
+    q_target_params = q_net.init(q_key, init_hidden_state, init_x)
+
+    # Make Mixer Network
+    dummy_agent_qs = jnp.zeros(
+        (
+            cfg.system.sample_batch_size,
+            cfg.system.sample_sequence_length - 1,
+            num_agents,
+        ),
+        "float32",
+    )
+    global_env_state_shape = (
+        env.observation_spec().generate_value().global_state[0, :].shape
+    )  # NOTE: Env wrapper currently duplicates env state for each agent
+    dummy_global_env_state = jnp.zeros(
+        (
+            cfg.system.sample_batch_size,
+            cfg.system.sample_sequence_length - 1,
+            *global_env_state_shape,
+        ),
+        "float32",
+    )
+    q_mixer = hydra.utils.instantiate(
+        cfg.network.mixer_network,
+        num_actions=action_dim,
+        num_agents=num_agents,
+        embed_dim=cfg.system.qmix_embed_dim,
+    )
+    mixer_online_params = q_mixer.init(q_key, dummy_agent_qs, dummy_global_env_state)
+    mixer_target_params = q_mixer.init(q_key, dummy_agent_qs, dummy_global_env_state)
+
+    # Pack params
+    params = QMIXParams(q_params, q_target_params, mixer_online_params, mixer_target_params)
+
+    # Optimiser
+    opt = optax.chain(
+        optax.adam(learning_rate=cfg.system.q_lr),
+    )
+    opt_state = opt.init((params.online, params.mixer_online))
+
+    # Distribute params, opt states and hidden states across all devices
+    params = replicate(params)
+    opt_state = replicate(opt_state)
+    init_hidden_state = replicate(init_hidden_state)
+
+    init_acts = env.action_spec().generate_value()
+
+    # NOTE: Term_or_trunc refers to the the joint done, ie. when all agents are done or when the
+    # episode has terminated. We use this exclusively in QMIX. Terminal refers to individual agent
+    # dones. We keep this in this file for consistency with IQL.
+    init_transition = Transition(
+        obs=init_obs,  # (A, ...)
+        action=init_acts,  # (A,)
+        reward=jnp.zeros((1,), dtype=float),
+        terminal=jnp.zeros((1,), dtype=bool),
+        term_or_trunc=jnp.zeros((1,), dtype=bool),
+        next_obs=init_obs,
+    )
+
+    # Initialise trajectory buffer
+    rb = fbx.make_trajectory_buffer(
+        # n transitions gives n-1 full data points
+        sample_sequence_length=cfg.system.sample_sequence_length,
```

Comment:
For consistency either make this `cfg.system.sample_sequence_length` or change IQL also there's a mention of this in the docs

Reply:
I saw the comment, but opted not to change any of this since I thought this is probably logic from flashbax. @lbeyers can you please comment on this?

Reply:
How you name/handle this depends on what you want the number you put in the config to mean

Reply:
But if you want "sample_sequence_length" to mean "the number of effective training samples I get in the time dimension", then you have to add the +1 back to the flashbax init